### PR TITLE
add xtcp library.

### PR DESCRIPTION
**Pull request by xfxdev**
Please check if what you want to add to `awesome-go` list meets [quality standards](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard) before sending pull request. Thanks!

**Please provide package links to:**
- github.com repo: https://github.com/xfxdev/xtcp
- godoc.org: https://godoc.org/github.com/xfxdev/xtcp
- goreportcard.com: https://goreportcard.com/report/github.com/xfxdev/xtcp
- coverage service link (gocover, coveralls etc.):  https://gocover.io/github.com/xfxdev/xtcp

**Note**: that new categories can be added only when there are 3 packages or more.

**Make sure that you've checked the boxes below before you submit PR:**
- [x] I have added my package in alphabetical order
- [x] I know that this package was not listed before
- [x] I have added godoc link to the repo and to my pull request
- [x] I have added coverage service link to the repo and to my pull request
- [x] I have added goreportcard link to the repo and to my pull request
- [x] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

Thanks for your PR, you're awesome! :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -727,6 +727,7 @@ See [go-hardware](https://github.com/rakyll/go-hardware) for a comprehensive lis
 * [tcp_server](https://github.com/firstrow/tcp_server) - A Go library for building tcp servers faster.
 * [utp](https://github.com/anacrolix/utp) - Go uTP micro transport protocol implementation.
 * [winrm](https://github.com/masterzen/winrm) - A Go WinRM client to remotely execute commands on Windows machines
+* [xtcp](https://github.com/xfxdev/xtcp) - A TCP Server Framework with simultaneous full duplex communication,graceful shutdown,custom protocal.
```
Comment (tmcarr) [inline README.md]: protocol

Comment (xfxdev) [inline README.md]: Its a framework to fast build tcp server.

Comment (tmcarr) [inline README.md]: No, I'm pointing out a typo. s/protocal/protocols/ in your change.

Comment (xfxdev) [inline README.md]: @tmcarr Fixed it just now. sorry.

Comment (tmcarr) [inline README.md]: Oh no big deal, just noticed the typo and thought I'd point it out. :)

**Pull request merged**